code/modules/reagents/chemistry/reagents/drugs.dm

```diff
@@ -297,38 +346,46 @@
 			M.bodytemperature -= 70
 	return list(effect, update_flags)
 
+// makes you faster, increases the duration of stuns, removes a LOT of stamina, makes you skinny, and does brain damage
 /datum/reagent/methamphetamine
 	name = "Methamphetamine"
 	id = "methamphetamine"
-	description = "Reduces stun times by about 300%, speeds the user up, and allows the user to quickly recover stamina while dealing a small amount of Brain damage. If overdosed the subject will move randomly, laugh randomly, drop items and suffer from Toxin and Brain damage. If addicted the subject will constantly jitter and drool, before becoming dizzy and losing motor control and eventually suffer heavy toxin damage."
+	description = "Increases stun times, speeds the user up, and allows the user to quickly recover stamina while dealing a large amount of brain damage and making the user waste away. If overdosed the subject will move randomly, laugh randomly, drop items and suffer from Toxin and Brain damage. If addicted the subject will constantly jitter and drool, before becoming dizzy and losing motor control and eventually suffer heavy toxin damage."
 	reagent_state = LIQUID
 	color = "#60A584" // rgb: 96, 165, 132
 	overdose_threshold = 20
 	addiction_chance = 10
 	addiction_threshold = 5
 	metabolization_rate = 0.6
+	addiction_decay_rate = 0.1 // very low, to prevent people from abusing the massive speed boost for too long. forces them to take long downtimes to not die from brain damage.
 	heart_rate_increase = 1
 	taste_description = "speed"
 
+/datum/reagent/methamphetamine/on_mob_add(mob/living/L)
+	ADD_TRAIT(L, TRAIT_GOTTAGOFAST, id)
+	if(ishuman(L))
+		var/mob/living/carbon/human/H = L
+		H.physiology.stun_mod *= 1.5 // takes 1.5 times longer to get up as they are off their head
+
 /datum/reagent/methamphetamine/on_mob_life(mob/living/M)
 	var/update_flags = STATUS_UPDATE_NONE
+	var/resent_consumption = holder.addiction_threshold_accumulated[type]
 	if(prob(5))
 		M.emote(pick("twitch_s","blink_r","shiver"))
-	if(current_cycle >= 25)
-		M.AdjustJitter(10 SECONDS)
-	M.AdjustDrowsy(-20 SECONDS)
-	M.AdjustParalysis(-5 SECONDS)
-	M.AdjustStunned(-5 SECONDS)
-	M.AdjustWeakened(-5 SECONDS)
-	update_flags |= M.adjustStaminaLoss(-2, FALSE)
+	M.AdjustJitter(10 SECONDS)
```

Comment:
wanna add a reasonable `upper_bound` arg?
Without it, for each second you go fast (and jitter), you would *also* spend 4 more seconds being jittery after the effect runs off. Not saying it's a bad thing necessarily, just might have been unintended.